control-plane/agents/src/bin/core/controller/reconciler/node/nexus.rs

```diff
@@ -0,0 +1,173 @@
+
+    let mut volumes_moved = 0;
+    let mut volumes_on_node = 0;
+
+    for volume in volumes.iter() {
+        let vol_spec = volume.spec();
+
+        if let Some(target) = vol_spec.target() {
+            if target.node() != node_id {
+                continue;
+            }
+            volumes_on_node += 1;
+            let nexus_id = target.nexus().clone();
+            let vol_id = vol_spec.uuid.clone();
+
+            let locked_volume_opt = context
+                .registry()
+                .specs()
+                .get_locked_volume(&vol_spec.uuid.clone());
+            match locked_volume_opt {
+                Some(v) => {
+                    match v.operation_guard() {
+                        Ok(mut guard) => {
+                            let config = vol_spec.config().clone();
+                            if config.is_none() {
+                                tracing::error!("Failed to get config");
+                                continue;
+                            }
+                            let frontend_nodes = config.unwrap().frontend().node_names();
+                            if frontend_nodes.is_empty() {
+                                tracing::error!("Cannot republish, volume has no application");
+                                continue;
+                            }
+                            let frontend_node_id: NodeId = frontend_nodes[0].clone().into();
+                            tracing::info!(
+                                volume.uuid = vol_id.as_str(),
+                                nexus.uuid = nexus_id.as_str(),
+                                node.id = node_spec.id().as_str(),
+                                "Moving volume"
+                            );
+                            if let Err(e) =
+                                republish_volume(&mut guard, context, &vol_id, &frontend_node_id)
+                                    .await
+                            {
+                                tracing::error!(
+                                    error=%e,
+                                    volume.uuid = vol_id.as_str(),
+                                    nexus.uuid = nexus_id.as_str(),
+                                    node.id = node_spec.id().as_str(),
+                                    "Failed to republish volume"
+                                );
+                                continue;
+                            }
+                            volumes_moved += 1;
+                            tracing::info!(
+                                volume.uuid = vol_id.as_str(),
+                                nexus.uuid = nexus_id.as_str(),
+                                node.id = node_spec.id().as_str(),
+                                "Moved volume"
+                            );
+                        }
+                        Err(e) => {
+                            tracing::error!(error=%e, "Failed to get operation guard");
```

Comment:
IIRC that's already traced automatically so we can simply continue

Reply:
Removed. Also the continue is redundant, so removed that as well.